From ae8c93fb73fa22398ed9bab1609abca595d68375 Mon Sep 17 00:00:00 2001
From: Sachin Mohan Gadag <sgadag@codeaurora.org>
Date: Tue, 4 Dec 2018 14:26:18 +0530
Subject: [PATCH] tinyalsa: Fix tinyplay runtime issue

- Declare config with default value, as set_sw_params
  is failing because of garbage value.
- Send number of frames read instead of bytes read to
  pcm_writei.

Change-Id: I615caf4a62376cd6d9b8abbd9fcfc29166e1c6c2
---
 git/utils/tinyplay.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/git/utils/tinyplay.c b/git/utils/tinyplay.c
index d0c1e36..448e81f 100644
--- a/git/utils/tinyplay.c
+++ b/git/utils/tinyplay.c
@@ -311,7 +311,7 @@ void print_usage(const char *argv0)
 
 int main(int argc, const char **argv)
 {
-    struct cmd cmd;
+    struct cmd cmd = {0};
     struct ctx ctx;
 
     if (argc < 2) {
@@ -409,7 +409,7 @@ int play_sample(struct ctx *ctx)
     do {
         num_read = fread(buffer, 1, size, ctx->file);
         if (num_read > 0) {
-            if (pcm_writei(ctx->pcm, buffer, num_read) < 0) {
+            if (pcm_writei(ctx->pcm, buffer, pcm_bytes_to_frames(ctx->pcm, num_read)) < 0) {
                 fprintf(stderr, "error playing sample\n");
                 break;
             }
-- 
2.18.0

